### PR TITLE
Insar template rendering

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: hyp3_metadata/__main__.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 from dateutil.parser import parse as dt_parse
 
 from hyp3_metadata import __version__, create_metadata_file_set_rtc
-from hyp3_metadata.util import SUPPORTED_DEMS, populate_example_data
+from hyp3_metadata.rtc import populate_example_data
```
Comment (jhkennedy) [inline hyp3_metadata/__main__.py]: hmm, handleing different products here might be annoying... Could be a good use-case for subparsers:

```
python -m hyp3_metadata rtc [RTC_OPTIONS]
python -m hyp3_metadata insar [INSAR_OPTIONS]
```
Or via the entrypoint game we setup in hyp3-gamma

File: hyp3_metadata/create.py [hunk 2]
```diff
@@ -20,6 +19,24 @@ def create_metadata_file_set_rtc(product_dir: Path, granule_name: str, dem_name:
         processor_name=processor_name,
         processor_version=processor_version,
     )
-    writer = RtcMetadataWriter(payload)
+    writer = rtc.RtcMetadataWriter(payload)
 
     return writer.create_metadata_file_set()
+
+
+def create_metadata_file_set_insar(product_dir, granule_name_ref, granule_name_sec, processing_date, looks, dem_name,
```
Comment (jhkennedy) [inline hyp3_metadata/create.py]: I think I like `ref_granule_name`, `sec_granule_name` better... but I don't feel strongly (just how you typically say it)

`reference_name` I think would also work for me... or `reference_granule_name`...

File: hyp3_metadata/insar.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import hyp3_metadata
+from hyp3_metadata import data
+from hyp3_metadata.util import get_granule_type, get_projection, get_thumbnail_encoded_string, render_template
+
+
+class InSarMetadataWriter:
+    def __init__(self, payload: dict):
+        self.payload = payload
+        self.dir = payload['product_dir']
+        self.file_name = payload['product_dir'].name
```
Comment (jhkennedy) [inline hyp3_metadata/insar.py]: `product_dir` and `product_name`?

File: hyp3_metadata/insar.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        info = gdal.Info(str(reference_file), format='json')
+        payload['pixel_spacing'] = info['geoTransform'][1]
+        payload['projection'] = get_projection(info['coordinateSystem']['wkt'])
+
+        payload['thumbnail_encoded_string'] = get_thumbnail_encoded_string(reference_file)
+
+        content = render_template(template, payload)
+        out_name = reference_file.name if not strip_ext else reference_file.stem
+        if name_only:
+            out_name = payload['product_dir'].name
```
Comment (jhkennedy) [inline hyp3_metadata/insar.py:114]: Both could be combined into a strip suffix option...

```python
strip_suffix = '.tif'
strip_suffix = '_vert_disp.tif'
```
but that'd require knowing the suffix which these don't. 

I want to be able to combine them somehow though...

File: hyp3_metadata/rtc.py [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
 from pathlib import Path
 from typing import List, Optional
 
 from osgeo import gdal
 
-from hyp3_metadata.util import get_dem_template_id, get_projection, get_thumbnail_encoded_string, render_template, \
-    strip_polarization
+import hyp3_metadata
+from hyp3_metadata import data
+from hyp3_metadata.util import get_granule_type, get_polarizations, get_projection, get_thumbnail_encoded_string, \
```
Comment (jhkennedy) [inline hyp3_metadata/rtc.py]: I'd just do `from hyp3_metadata import util` here

File: hyp3_metadata/rtc.py [hunk 6]
```diff
@@ -113,3 +118,54 @@ def create_metadata_file(cls, payload: dict, template: str, reference_file: Path
...
+    if dem_name.startswith('IFSAR'):
+        return 'ifsar'
+    if dem_name.startswith('NED'):
+        return 'ned'
+    if dem_name.startswith('REMA'):
+        return 'rema'
+    if dem_name.startswith('SRTM'):
+        return 'srtm'
+    if dem_name == 'GLO-30':
+        return 'cop'
```
Comment (jhkennedy) [inline hyp3_metadata/rtc.py:141]: This feels more like a `util` function even though RTC is the only thing leveraging different DEMs

File: hyp3_metadata/templates/rtc/readme.md.txt.j2 [hunk 7]
```diff
@@ -25,6 +25,7 @@ l:          Entire Area (e) or Clipped Area (c)
 m:          Dead Reckoning (d) or DEM Matching (m)
 ssss:       Product ID
 ```
+```
```
Comment (jhkennedy) [inline hyp3_metadata/templates/rtc/readme.md.txt.j2]: duplicate?

File: hyp3_metadata/util.py [hunk 8]
```diff
@@ -1,28 +1,16 @@
...
+from typing import Dict, Tuple
 
 from PIL import Image
 from jinja2 import Environment, PackageLoader, StrictUndefined, select_autoescape
 from osgeo import osr
 
-import hyp3_metadata
-from hyp3_metadata import data
-
 SUPPORTED_DEMS = ['EU_DEM_V11', 'GIMP', 'IFSAR', 'NED13', 'NED1', 'NED2', 'REMA', 'SRTMGL1', 'SRTMGL3', 'GLO-30']
```
Comment (jhkennedy) [inline hyp3_metadata/util.py]: This should move into RTC if we're moving `get_dem_template_id` there

In that case, maybe both should just be class methods of the product writer

**Review by jhkennedy — COMMENTED** [review]
Overall, I think this looks pretty good

File: hyp3_metadata/__main__.py (hunk 1 above)
Comment (Jlrine2) [reply hyp3_metadata/__main__.py]: Yeah, I haven't gotten to dealing with main yet, That seems like a good solution, I just haven't looked at it yet.

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**